Source/SourceKittenFramework/SourceKitVariant.swift

```diff
@@ -0,0 +1,345 @@
+                }
+                return anyDictionary
+            case let .string(string):
+                return string
+            case let .int64(int64):
+                return Int(int64)
+            case let .bool(bool):
+                return bool
+            case let .uid(uidString):
+                return uidString
+            case .none:
+                return nil
+            }
+        }
+    }
+
+    fileprivate final class _ResponseBox {
+        private let response: sourcekitd_response_t
+        init(_ response: sourcekitd_response_t) {
+            self.response = response
+        }
+        deinit {
+            sourcekitd_response_dispose(response)
+        }
+    }
+
+    private init(variant: sourcekitd_variant_t, response: _ResponseBox) {
+        box = _VariantBox(variant: .variant(variant, response))
+    }
+
+}
+
+// MARK: - Equatable with String
+public func ==(lhs: SourceKitVariant?, rhs: String) -> Bool {
+    return lhs.map { $0 == rhs } ?? false
+}
+
+public func ==(lhs: String, rhs: SourceKitVariant?) -> Bool {
+    return rhs == lhs
+}
+
+extension SourceKitVariant {
+    public static func ==(lhs: SourceKitVariant, rhs: String) -> Bool {
+        return lhs.string.map { $0 == rhs } ?? false
+    }
+
+    public static func ==(lhs: String, rhs: SourceKitVariant) -> Bool {
+        return rhs == lhs
+    }
+}
+
+// MARK: - Equatable with RawRepresentable
+public func ==<T: RawRepresentable>(lhs: SourceKitVariant?, rhs: T) -> Bool
+    where T.RawValue == String {
+        return lhs == rhs.rawValue
+}
+
+public func ==<T: RawRepresentable>(lhs: T, rhs: SourceKitVariant?) -> Bool
+    where T.RawValue == String {
+        return rhs == lhs
+}
+
+extension SourceKitVariant {
+    public static func ==<T: RawRepresentable>(lhs: SourceKitVariant, rhs: T) -> Bool
+        where T.RawValue == String {
+            return lhs == rhs.rawValue
+    }
+
+    public static func ==<T: RawRepresentable>(lhs: T, rhs: SourceKitVariant) -> Bool
+        where T.RawValue == String {
+            return rhs == lhs
+    }
+}
+
+// MARK: - SwiftDocKey
+extension SourceKitVariant {
+    var annotatedDeclaration: String? { return self[.annotatedDeclaration]?.string }
+    var bodyLength: Int? { return self[.bodyLength]?.int }
+    var bodyOffset: Int? { return self[.bodyOffset]?.int }
+    var diagnosticStage: String? { return self[.diagnosticStage]?.string }
+    var filePath: String? { return self[.filePath]?.string }
+    var fullXMLDocs: String? { return self[.fullXMLDocs]?.string }
+    var kind: SourceKitVariant? { return self[.kind] }
+    var length: Int? { return self[.length]?.int }
+    var name: String? { return self[.name]?.string }
+    var nameLength: Int? { return self[.nameLength]?.int }
+    var nameOffset: Int? { return self[.nameOffset]?.int }
+    var offset: Int? { return self[.offset]?.int }
+    var substructure: [SourceKitVariant]? { return self[.substructure]?.array }
```

Comment:
I'm considering that `substructure` would returns non optional.
```diff
diff --git a/Source/SourceKittenFramework/SourceKitVariant.swift b/Source/SourceKittenFramework/SourceKitVariant.swift
index e8e5ee5..16641e1 100644
--- a/Source/SourceKittenFramework/SourceKitVariant.swift
+++ b/Source/SourceKittenFramework/SourceKitVariant.swift
@@ -332,10 +332,10 @@ extension SourceKitVariant {
     var nameLength: Int? { return self[.nameLength]?.int }
     var nameOffset: Int? { return self[.nameOffset]?.int }
     var offset: Int? { return self[.offset]?.int }
-    var substructure: [SourceKitVariant]? { return self[.substructure]?.array }
-    var syntaxMap: [SourceKitVariant]? { return self[.syntaxMap]?.array }
+    var substructure: [SourceKitVariant] { return self[.substructure]?.array ?? [] }
+    var syntaxMap: [SourceKitVariant] { return self[.syntaxMap]?.array ?? [] }
     var typeName: String? { return self[.typeName]?.string }
-    var inheritedtypes: [SourceKitVariant]? { return self[.inheritedtypes]?.array }
+    var inheritedtypes: [SourceKitVariant] { return self[.inheritedtypes]?.array ?? [] }
 }
 
 // MARK: - Custom
```
With current implementation:
```swift
variant.substructure.map { $0 in // $0 is [SourceKitVariant] }
variant.substructure?.map { $0 in // $0 is SourceKitVariant }
```
will be changed to:
```swift
variant.substructure.map { $0 in // $0 is SourceKitVariant }
variant.substructure?.map { $0 in } // invalid
```
Checking existence of `key.substructure` can be achieved by another way: `variant[.substructure] != nil`

I think those properties are convenience accessor, and should be more convenient.